cli/internal/check/diff.go

```diff
@@ -499,14 +514,45 @@ func checkNewUpdateIfDocuments(glob string, documents []*docs.FilteredDoc, filte
 func checkNewUpdateIfSections(glob string, sections []docs.FilteredSection, filter config.DocumentationFilter, cb updateResultMapCallback, action string, check DiffCheck) {
 	for _, section := range sections {
 		if docs.SectionMatches(section.Section.ID, section.Section.DocumentID, section.Section.SourceID, section.Tags, &filter, false) {
-			cb(section.Section.SourceID+"/"+section.Section.DocumentID+"#"+section.Section.ID,
+			uri := docs.DocumentURI{SourceID: section.Section.SourceID, DocumentPath: section.Section.DocumentID, Section: section.Section.ID}
+			cb(uri.String(),
 				fmt.Sprintf("Update this document if any files matching `%s` were %s (matching file: %s).", glob, action, check.File),
 				check)
 		}
 		checkNewUpdateIfSections(glob, section.Sections, filter, cb, action, check)
 	}
 }
 
+func buildValidIDMap(documents []*docs.FilteredDoc) map[string]bool {
+	validIDs := make(map[string]bool)
```

Comment:
nit: You might want to consider using a `map[string]struct{}` and then doing an existence check rather than using a boolean to be a bit more efficient. I'm good with it as is as well.